### PR TITLE
Changes to StoreInvoice client and result

### PR DESCRIPTION
**Pull request by ndeet**
- add invoice status to client
- specific classes for QRCode types
- amount as string and addding currency on createInvoice

### CODE REVIEW

File: src/Client/StoreInvoice.php [hunk 1]
```diff
@@ -4,17 +4,18 @@
 
 namespace NodelessIO\Client;
 
-use NodelessIO\Result\StoreInvoice as ResultInvoice;
+use NodelessIO\Result\StoreInvoice as ResultStoreInvoice;
```
Comment (utxo-one) [inline src/Client/StoreInvoice.php]: Let's follow the Laravel convention and always put the "type" at the end, so StoreInvoiceResult, same on other methods.

**Review by utxo-one — CHANGES_REQUESTED** [review]

File: src/Client/StoreInvoice.php (hunk 1 above)
Comment (utxo-one) [reply src/Client/StoreInvoice.php]: In fact I think Response is a better word than Result, what do you think?

Comment (ndeet) [reply src/Client/StoreInvoice.php]: Yeah sounds better, so the class is then `StoreInvoiceResponse` and the namespace is `NodelessIO\Response`?

Should I do the same for the client classes so the classname becomes `StoreInvoiceClient`?

Comment (utxo-one) [reply src/Client/StoreInvoice.php]: Yes to both please

Comment (ndeet) [reply src/Client/StoreInvoice.php]: Changed and also adjusted to latest merges on checkoutLink and fiat support.

**Review by utxo-one — APPROVED** [review]

**Pull request merged**